### PR TITLE
[FIX] mrp: action_unbuild

### PR DESCRIPTION
**Pull request by simongoffin**
When creating the stock.move.lots from the quants, in action_unbuild,
in some cases, it is possible to have several quants for the same product
with different lot_id. So before the fix, it raised an error because original
was expected to be a singleton. Now all the quants are grouped and summed by lot
to create one stock.move.lots per lot.

### CODE REVIEW

File: addons/mrp/models/mrp_unbuild.py [hunk 1]
```diff
@@ -119,12 +119,19 @@ def action_unbuild(self):
...
                 original = original_quants.filtered(lambda quant: quant.product_id == produce_move.product_id)
-                self.env['stock.move.lots'].create({
-                    'move_id': produce_move.id,
-                    'lot_id': original.lot_id.id,
-                    'quantity_done': produce_move.product_uom_qty,
-                    'quantity': produce_move.product_uom_qty
-                })
+                group_by_lot = {}
+                for o in original:
+                    if o.lot_id.id in group_by_lot:
```
Comment (mart-e) [inline addons/mrp/models/mrp_unbuild.py]: You can replace these 4 lines by
`group_by_lot.setdefault(o.lot_id.id, 0) += produce_move.product_uom_qty`

Comment (simongoffin) [inline addons/mrp/models/mrp_unbuild.py]: Alright thanks!

File: addons/mrp/models/mrp_unbuild.py [hunk 2]
```diff
@@ -119,12 +119,16 @@ def action_unbuild(self):
...
                 original = original_quants.filtered(lambda quant: quant.product_id == produce_move.product_id)
-                self.env['stock.move.lots'].create({
-                    'move_id': produce_move.id,
-                    'lot_id': original.lot_id.id,
-                    'quantity_done': produce_move.product_uom_qty,
-                    'quantity': produce_move.product_uom_qty
-                })
+                group_by_lot = {}
+                for o in original:
+                    group_by_lot.setdefault(o.lot_id.id, 0) += produce_move.product_uom_qty
```
Comment (jco-odoo) [inline addons/mrp/models/mrp_unbuild.py]: += o.qty would be more logical

Comment (jco-odoo) [inline addons/mrp/models/mrp_unbuild.py]: as the move lots are linked to the same move, the quantity on the move is the sum of the move lots.  So, if you put multiple move lots, that would go against that.

**Comment by mlaitinen** [comment]
Can you make the CI run again? Or is there anything else that prevents this from being merged? Our client just encountered this error while unbuilding a lot with several quants.

**Comment by mlaitinen** [comment]
Just in case anyone gets as confused as I got, the PR was closed because this issue was fixed in 671c20555e97fa882dfc42cbd9e429141a87e7c5.

**Pull request closed without merging**